### PR TITLE
Replace all rule tokens in prompt message 

### PR DESCRIPTION
**Pull request by Joe-Soap**
## Scope 
Affects all prompt text values that use replacement tokens

## Description
Within the prompt property method of the form module, I updated the replace calls, where it replaces contextual tokens relating to the rules definition,  to use a regular expression of the same respective values in order to have it replace all tokens present in the prompt text. 

The tokens used are the following: 
* identifier
* name
* ruleValue

If one uses a singular token multiple times in the prompt, for example,

`{ 
    type: 'levelCheck[2]', 
    prompt: 'Level {ruleValue} not detected. Please ensure that {name} is set to a level at least equal to {ruleValue}.' 
}`

 the first token of {ruleValue} is successfully replaced, but the subsequent {ruleValue} isn't. This change merely applies a regex with the "global" flag set in order to replace multiple tokens of the same name found in the prompt text. 

### Broken
https://jsfiddle.net/08361wud/3/

#### Fixed
https://jsfiddle.net/1wyx2vt7/

#### Broken
image

#### Fixed
image

The above example illustrates a use case and the problem.

### CODE REVIEW

File: src/definitions/behaviors/form.js [hunk 1]
```diff
@@ -545,18 +545,18 @@ $.fn.form = function(parameters) {
               name
             ;
             if(requiresValue) {
-              prompt = prompt.replace('{value}', $field.val());
+              prompt = prompt.replace(/\{value\}/g $field.val());
```
Comment (lubber-de) [inline src/definitions/behaviors/form.js]: Missing comma, should read
```javascript
prompt = prompt.replace(/\{value\}/g, $field.val());
```

Comment (Joe-Soap) [reply src/definitions/behaviors/form.js]: fixed. Thank you

**Comment by lubber-de** [comment]
And, as @ColinFrick already pointed out: Name your PR Title properly instead of "Develop". Thank you 🙂

**Comment by Joe-Soap** [comment]
> All changes have to be made in the source files, please do not commit the dist files.
> Additionally please change the title to something meaningful

Done

**Comment by Joe-Soap** [comment]
> And, as @ColinFrick already pointed out: Name your PR Title properly instead of "Develop". Thank you 🙂

Done

**Comment by Joe-Soap** [comment]
> Please, still, remove the dist files. You may have to reset your develop branch, apply your patch to the single form.js in the behavior folder _only_ and then commit and do a force push

I reverted the changes to the dist folder. Do I really need to reset?

File: src/definitions/behaviors/form.js [hunk 2]
```diff
@@ -545,18 +545,18 @@ $.fn.form = function(parameters) {
...
+              prompt = prompt.replace(/\{value\}/g, $field.val());
             }
             if(requiresName) {
               $label = $field.closest(selector.group).find('label').eq(0);
               name = ($label.length == 1)
                 ? $label.text()
                 : $field.prop('placeholder') || settings.text.unspecifiedField
               ;
-              prompt = prompt.replace('{name}', name);
+              prompt = prompt.replace(/\{name\}/g', name);
```
Comment (lubber-de) [inline src/definitions/behaviors/form.js]: Remove the single quote, which breaks the script. It should read:
```javascript
prompt = prompt.replace(/\{name\}/g, name);
```

Comment (Joe-Soap) [reply src/definitions/behaviors/form.js]: Fixed. Apologies. This is my first contribution. Silly mistake. Nerves, I think.

Comment (lubber-de) [reply src/definitions/behaviors/form.js]: Just test your code before submitted a PR 😉 don't do "dry" PRs, because you think it will work

Comment (Joe-Soap) [reply src/definitions/behaviors/form.js]: Understood. Agreed. That's one of the underlying reasons I was apologizing

**Comment by Joe-Soap** [comment]
> LGTM
> I adjusted the PR description and added another testcase where the fix is applied and proof screenshots

Thanks for your help lubbe

**Comment by Joe-Soap** [comment]
@ColinFrick After the recent changes, is there anything else for me to change? The merge will only be one file...

**Pull request merged**